backend/src/email/services/email-template.service.ts

```diff
@@ -234,16 +235,41 @@ const addToMessageLogs = async (
   }
 }
 
+/**
+ * Checks for invalid email recipients from an array of records
+ *
+ * @param records
+ */
 const hasInvalidEmailRecipient = (
   records: MessageBulkInsertInterface[]
 ): boolean => {
   return records.some((record) => !validator.isEmail(record.recipient))
 }
 
+/**
+ * Attempts to hydrate the first record.
+ * @param records
+ * @param templateBody
+ * @param templateSubject - optional
```

Comment:
I am sure there are ways to reflect optional params but idk which style guide are we following now